File: backend/app/schemas/quiz.py
from datetime import datetime

from pydantic import BaseModel, Field


class QuizGenerateRequest(BaseModel):
    topic: str = ""
    num_questions: int = Field(default=5, ge=1, le=20)


class QuizResponse(BaseModel):
    id: int
    document_id: int
    topic: str
    questions: list[dict]
    created_at: datetime


class QuizAttemptCreate(BaseModel):
    answers: list[int]


class QuizAttemptResponse(BaseModel):
    id: int
    quiz_id: int
    answers: list[int]
    score: int
    total: int
    created_at: datetime

File: frontend/src/lib/quiz.ts
export interface QuizQuestion {
  question: string;
  options: string[];
  answer_index: number;
  explanation: string;
}

export interface Quiz {
  id: number;
  document_id: number;
  topic: string;
  questions: QuizQuestion[];
  created_at: string;
}

export interface QuizAttempt {
  id: number;
  quiz_id: number;
  answers: number[];
  score: number;
  total: number;
  created_at: string;
}

export function scoreTier(pct: number): "high" | "mid" | "low" {
  if (pct >= 80) return "high";
  if (pct >= 50) return "mid";
  return "low";
}
